File: content-solutions/stack/lightest-box.java
import java.util.*;

class Solution {
    public List<Integer> solution(String[] commands) {
        Deque<Integer> stack = new ArrayDeque<>();
        Deque<Integer> mins = new ArrayDeque<>();
        List<Integer> answer = new ArrayList<>();
        for (String command : commands) {
            String[] parts = command.split(" ");
            if (parts[0].equals("push")) {
                int w = Integer.parseInt(parts[1]);
                stack.push(w);
                mins.push(mins.isEmpty() ? w : Math.min(w, mins.peek()));
            } else if (parts[0].equals("pop")) {
                if (!stack.isEmpty()) {
                    stack.pop();
                    mins.pop();
                }
            } else {
                answer.add(mins.isEmpty() ? -1 : mins.peek());
            }
        }
        return answer;
    }
}

File: content-solutions/stack/lightest-box.py
def solution(commands):
    stack = []
    mins = []
    answer = []
    for command in commands:
        parts = command.split()
        if parts[0] == "push":
            w = int(parts[1])
            stack.append(w)
            mins.append(w if not mins else min(w, mins[-1]))
        elif parts[0] == "pop":
            if stack:
                stack.pop()
                mins.pop()
        else:
            answer.append(mins[-1] if mins else -1)
    return answer
